### ISSUE TITLE
Add support for custom rule templates via configuration

### ISSUE THREAD

**Issue by dmzoneill** [ansible/ansible-lint#4957]
### Description
Currently, while ansible-lint allows disabling rules and configuring their behavior, there is no straightforward way to define custom organization-specific linting rules without writing Python code and packaging them as plugins. This creates a barrier for teams who want to enforce internal best practices.

### Proposed Solution
Add support for defining simple custom rules via the ansible-lint configuration file (.ansible-lint) using a template-based approach. For example:

```yaml
custom_rules:
  - id: org-no-shell-module
    description: Prefer command module over shell when possible
    pattern: '- name: .* shell:'
    severity: warning
    message: Consider using the command module instead of shell for better security
  
  - id: org-require-tags
    description: All tasks must have tags
    check: task_has_tags
    severity: error
```

### Use Case
Many organizations have internal standards that go beyond general Ansible best practices, such as:
- Requiring specific naming conventions for tasks
- Enforcing use of organization-approved modules
- Mandating certain variables or tags be present
- Prohibiting deprecated internal patterns

Currently, enforcing these requires either:
1. Writing custom Python-based ansible-lint plugins (high barrier)
2. Using external tools alongside ansible-lint (fragmented workflow)
3. Manual code review (error-prone)

### Benefits
- Lower barrier to entry for teams to codify internal standards
- Keep all linting configuration in one place
- Reduce need for custom tooling
- Enable gradual adoption of team-specific rules

This would make ansible-lint more extensible for enterprise use cases while maintaining its focus on best practices.
